### DISCUSSION TITLE
Kodi sets watched status for videos downloaded later the same day

### DISCUSSION THREAD

**Discussion by cbunn81** [jmbannon/ytdl-sub#603] [Q&A] [answered]
When using the preset `kodi_tv_show_by_date`, episodes are numbered by the date (e.g. using `s2023e0501` for today). In general, this doesn't present a problem for multiple videos from the same channel/playlist released on the same date, since the titles will be unique. But I have noticed an issue with watched status in Kodi.

If I watch a video with today's date in the episode number, then ytdl-sub scrapes more episodes from the same channel/playlist released on the same date, when those episodes are loaded by Kodi, it will mistakenly mark them as watched.

I'm assuming this is an issue with Kodi not being designed with multiple episodes of the same number in mind. So it's not really something for ytdl-sub to fix. And it's not a critical issue, as it's not that often that the necessary circumstances are met. But it's still something that I'd like to work out.

So first, I'm curious to know if anyone else has noticed this. And then if there's a good workaround that might be employed. I've been looking at the other naming schemes available and the only pre-built one that could work would be `season_by_year__episode_by_download_index`. According to [the comments on the definition of `download_index`](https://github.com/jmbannon/ytdl-sub/blob/a2f1a2ba949263f43d455bbfcb89f3dc140b87f3/src/ytdl_sub/entries/variables/entry_variables.py#L428), this value is set dynamically as entries are downloaded. So I'm wondering if that would be impacted by old episodes being deleted.

I also wonder if it might be possible to combine the two, as I generally find numbering episodes by date to be helpful. But perhaps a padded index could be included after the date.

**Comment by jmbannon** [comment, marked as answer]
@cbunn81 What version of ytdl-sub are you on? Maybe you are using an old version since this was added a bit later. In the TV show presets, there should be a 'date index' that ensures your episode numbers do not overlap when there are multiple eps in a single day.

i.e. two vids on 2023-03-15 should be:
s2023e031501
s2023e031502

**Reply by cbunn81** [reply]
Interesting! I didn't realize that. I'm on version 2023.04.29+a2f1a2b, as I have my script automatically update before scanning.

I was following the instructions from the [wiki](https://github.com/jmbannon/ytdl-sub/wiki/6.-Modifying-Your-Config-for-Your-Media-Player) and didn't notice that "date index" was different than "day" when checking out the other prebuilt presets. But I'll definitely give those a try.

I'm assuming that ytdl-sub won't attempt to rename files previously downloaded, right? Changing my config should only affect newly-downloaded files, I would imagine.

**Reply by jmbannon** [reply]
Yes, use the prebuilt presets:
https://github.com/jmbannon/ytdl-sub/wiki/8.-Using-Prebuilt-Presets-and-our-Example-Presets
https://ytdl-sub.readthedocs.io/en/latest/presets.html

The wiki is only meant to walk you through how it works - prebuilt presets are what you should be using for downloading stuff as TV shows. This will have the date index stuff all automatically added.

As for automatically updating - that functionality is WIP, so unfortunately the best work-around is to re-download everything using them.

**Reply by cbunn81** [reply]
Thank you. I can confirm that the functionality you mentioned in the pre-built presets addresses my initial issue.

I don't need the naming of the old episodes to be updated. It doesn't seem to be a problem for Kodi, and those old episodes will get deleted eventually.
